lewis/examples/modbus_device/__init__.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# *********************************************************************
+# lewis - a library for creating hardware device simulators
+# Copyright (C) 2016 European Spallation Source ERIC
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# *********************************************************************
+
+from lewis.devices import Device
+from lewis.adapters.modbus import ModbusAdapter, ModbusDataBank
+
+
+class ModbusDevice(Device):
+    pass
+
+
+class ModbusInterface(ModbusAdapter):
+    """
+    The class attributes di, co, ir and hr represent Discrete Inputs, Coils, Input Registers and
+    Holding Registers, respectively. Each attribute should be assigned a ModbusDataBank instance
+    by the Interface implementation.
+
+    Here, two basic ModbusDataBanks are created and initialized to a default value across the full
+    range of valid addresses. One DataBank is shared by di and co, and the other by ir and hr to
+    demonstrate overlaid memory segments. If you want each segment to have its own memory, just
+    create separate instances for all four.
+    """
+    di = ModbusDataBank.create_basic(False)
+    co = di
+    ir = ModbusDataBank.create_basic(0)
+    hr = ir
```

Comment:
Yes, it would break our separation, that is true, so maybe not a good way to go after all. I think however that it urgently shows the need for proper bindings, which brings me back to the original comment...how do we actually advertise this until we have such bindings?

In the generated documentation (the `rst`-files still need to be added under `docs/api/adapters` so that this gets generates) this should be marked as experimental with a small section describing how this works for now. One paragraph with "this is how you could modify the `Interface` in the example to achieve X instead" would probably enough.

Speaking of documentation, the release notes should also be updated. I also keep forgetting it as the procedure is new :)